### PR TITLE
fix Issue 16197 - Constructors/postblits and destructors don't match …

### PR DESCRIPTION
**Pull request by WalterBright**
…up for array initialisation

### CODE REVIEW

File: test/runnable/sdtor.d [hunk 1]
```diff
@@ -4154,6 +4154,25 @@ int test14815()
 static assert(test14815());
 
 /**********************************/
+// https://issues.dlang.org/show_bug.cgi?id=16197
+
+struct Elem {
+    static string r;
+    int x = -1;
+    this(this) { r ~= 'p'; printf("POSTBLIT %d\n", x++); }
+    ~this()    { r ~= 'd'; printf("DTOR %d\n"    , x++); }
```
Comment (andralex) [inline test/runnable/sdtor.d]: One unpleasant consequence of this idiom: calling `~=` in a destructor causes reallocation which may cause garbage collection, which aborts the program...

Comment (WalterBright) [inline test/runnable/sdtor.d]: True, but this is a test case only, and in this case the destructors are not being called by the gc. So it's all right here.

Comment (andralex) [inline test/runnable/sdtor.d]: The problem is real - it did occur in some of Lucia's code.

Comment (WalterBright) [inline test/runnable/sdtor.d]: I know it is real, but not for this test case.

**Pull request merged**